### PR TITLE
task 868-871 SQL and MSMARCO

### PR DESCRIPTION
**Pull request by ujjwalaananth**

### CODE REVIEW

**Comment by ujjwalaananth** [comment]
@danyaljj Update on previous failing tests: fixed.

**Comment by danyaljj** [comment]
@ghlai9665 Wondering if you can review this PR?

**Comment by garyhlai** [comment]
> @ghlai9665 Wondering if you can review this PR?

@danyaljj of course. I can take a look on Tuesday afternoon / night. Is that ok with you?

**Comment by aarunku5** [comment]
Thanks, a couple of corrections:

869-
In the definition you've mentioned true/false, but your outputs are 1/0, make them uniform.

870-
Change the first negative example to a different passage/question.
Also, can you ensure proper capitalization/grammar in the queries? There are a few instances where it's missing.

871- 
The use of the word 'question' in the definition is a bit misleading since the actual output is more like a query term you'd put into a search engine. Update the definition to make that clear.

**Comment by ujjwalaananth** [comment]
> Thanks, a couple of corrections:
> 
> 869- In the definition you've mentioned true/false, but your outputs are 1/0, make them uniform.
> 
> 870- Change the first negative example to a different passage/question. Also, can you ensure proper capitalization/grammar in the queries? There are a few instances where it's missing.
> 
> 871- The use of the word 'question' in the definition is a bit misleading since the actual output is more like a query term you'd put into a search engine. Update the definition to make that clear.

Not sure what you meant about the capitalization issues in 870? All other suggestions have been implemented. @aarunku5

**Comment by aarunku5** [comment]
Thanks, to clarify:

870
For example, in P2, your query is "how long does broken ribs take to heal"- so are most queries in this format across instances? If so, you need to include in your definition that the queries may not follow proper grammatical structure or be complete sentences. If not, capitalization/grammar fixes are needed on the instances where it is missing.

871
Still see the word 'question' in the definitions, please fix that.

**Comment by ujjwalaananth** [comment]
> Thanks, to clarify:
> 
> 870 For example, in P2, your query is "how long does broken ribs take to heal"- so are most queries in this format across instances? If so, you need to include in your definition that the queries may not follow proper grammatical structure or be complete sentences. If not, capitalization/grammar fixes are needed on the instances where it is missing.
> 
> 871 Still see the word 'question' in the definitions, please fix that.

extended description for 870, replaced occurrence in 871. If there's anything I've missed, please let me know.

**Comment by aarunku5** [comment]
Thanks, @danyaljj looks good to me

File: tasks/task868_cfq_mcd1_explanation_to_sql.json [hunk 1]
```diff
@@ -0,0 +1,2422 @@
...
+    "Positive Examples": [
+        {
+            "input": "Did M0 's star , director , art director , producer , and cinematographer direct M1 and direct M2 , M3 , and M4",
+            "output": "SELECT count(*) WHERE {\n?x0 ns:film.actor.film/ns:film.performance.film M0 .\n?x0 ns:film.cinematographer.film M0 .\n?x0 ns:film.director.film M0 .\n?x0 ns:film.director.film M1 .\n?x0 ns:film.director.film M2 .\n?x0 ns:film.director.film M3 .\n?x0 ns:film.director.film M4 .\n?x0 ns:film.film_art_director.films_art_directed M0 .\n?x0 ns:film.producer.film|ns:film.production_company.films M0\n}",
```
Comment (danyaljj) [inline tasks/task868_cfq_mcd1_explanation_to_sql.json:20]: What are the `\n?x0` characters?

**Comment by danyaljj** [comment]
Please make sure to set the `"Categories": [""]` and `"Domains": [""]` fields according to [the taxonomy presented here](https://github.com/allenai/natural-instructions-expansion/blob/master/doc/task-hierarchy.md).

**Comment by swarooprm** [comment]
@ujjwalaananth, can you answer reviewer questions so that we can go ahead with this PR?

**Comment by ujjwalaananth** [comment]
Added categories. Domain and language added for answer generation and question generation tasks, but not for SQL-based tasks, as I was not sure of the relevant domain/language.

File: tasks/task868_cfq_mcd1_explanation_to_sql.json (hunk 1 above)
Comment (ujjwalaananth) [reply tasks/task868_cfq_mcd1_explanation_to_sql.json:20]: I believe ?x0 and ?x1 are placeholders for variables. \n indicates nextline escape sequence.

Comment (danyaljj) [reply tasks/task868_cfq_mcd1_explanation_to_sql.json:20]: I see. Do think it would be more clear if we explain/exemplify it?

**Comment by aarunku5** [comment]
@swarooprm can you take a look at this?

File: tasks/task868_cfq_mcd1_explanation_to_sql.json (hunk 1 above)
Comment (ujjwalaananth) [reply tasks/task868_cfq_mcd1_explanation_to_sql.json:20]: Like mentioning in the explanation about the placeholder characters? I'm not entirely sure what you mean by exemplifying it.

**Pull request merged**